.github/workflows/python_package.yml

```
@@ -0,0 +1,89 @@
+name: Python-package
+
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+    branches:
+    - master
+
+env:
+  CONDA_ENV: test-env
+  GITHUB_ACTIONS: 'true'
+
+jobs:
+  test:
+    name: ${{ matrix.task }} (${{ matrix.os }}, ${{ matrix.compiler }}, Python ${{ matrix.python_version }})
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 60
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: macOS-latest
+            task: regular
+            python_version: 3.7
+            compiler: gcc
+          - os: macOS-latest
+            task: sdist
+            python_version: 3.8
+            compiler: gcc
+          - os: macOS-latest
+            task: bdist
+            python_version: 3.8
+            compiler: gcc
+          - os: macOS-latest
+            task: if-else
+            python_version: 3.8
+            compiler: gcc
+          - os: macOS-latest
+            task: mpi
+            method: source
+            python_version: 3.8
+            compiler: gcc
+          - os: macOS-latest
+            task: mpi
+            method: pip
+            python_version: 3.7
+            compiler: gcc
+          - os: macOS-latest
+            task: mpi
+            method: wheel
+            python_version: 3.7
+            compiler: gcc
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2.3.4
+        with:
+          fetch-depth: 5
+          submodules: true
+      - name: Setup and run tests
+        shell: bash
+        run: |
+          export TASK="${{ matrix.task }}"
+          export METHOD="${{ matrix.method }}"
+          export COMPILER="${{ matrix.compiler }}"
+          export PYTHON_VERSION="${{ matrix.python_version }}"
+          if [[ "${{ matrix.os }}" == "macOS-latest" ]]; then
+              export OS_NAME="macos"
+          elif [[ "${{ matrix.os }}" == "ubuntu-latest" ]]; then
+              export OS_NAME="linux"
+          fi
+          export AMDAPPSDK_PATH=$HOME/AMDAPPSDK
+          export LD_LIBRARY_PATH="$AMDAPPSDK_PATH/lib/x86_64:$LD_LIBRARY_PATH"
+          export LD_LIBRARY_PATH="/usr/local/clang/lib:$LD_LIBRARY_PATH"  # fix error "libomp.so: cannot open shared object file: No such file or directory" on Linux with Clang
+          export OPENCL_VENDOR_PATH=$AMDAPPSDK_PATH/etc/OpenCL/vendors
+          export BUILD_DIRECTORY="$GITHUB_WORKSPACE"
```

We don't run GPU jobs on macOS, so these `export`s are not needed.
```suggestion
          export BUILD_DIRECTORY="$GITHUB_WORKSPACE"
```

---

oh great